var/spack/repos/builtin/packages/lulesh/package.py

```diff
@@ -23,33 +23,66 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 ##############################################################################
 from spack import *
-import os
 
 
-class Lulesh(Package):
-    """Livermore Unstructured Lagrangian Explicit Shock Hydrodynamics (LULESH)
+class Lulesh(MakefilePackage):
+    """LULESH is a highly simplified application, hard-coded to only
+    style typical in scientific C or C++ based applications. Hard
+    code to only solve a Sedov blast problem with analytic answer
     """
-
+    tags = ['proxy-app']
     homepage = "https://codesign.llnl.gov/lulesh.php"
     url      = "https://codesign.llnl.gov/lulesh/lulesh2.0.3.tgz"
 
-    version("2.0.3", "336644a8750f71c7c6b9d2960976e7aa")
+    version('2.0.3', '336644a8750f71c7c6b9d2960976e7aa')
 
-    patch("remove_defaults.patch")
+    variant('mpi', default=True, description='Build with MPI support')
+    variant('openmp', default=True, description='Build with OpenMP support')
+    variant('visual', default=False,
+        description='Build with Visualization support (Silo, hdf5)')
 
-    variant('mpip', default=False)
+    depends_on('mpi', when='+mpi')
+    depends_on('silo', when='+visual')
+    depends_on('hdf5', when='+visual')
 
-    depends_on("mpi", type="build")
-    depends_on("mpip", when="+mpip")
+    @property
+    def build_targets(self):
+        targets = []
+        cxxflag = ' -g -O3 -I. '
+        ldflags = ' -g -O3 '
 
-    def install(self, spec, prefix):
-        if '+mpip' in spec:
-            os.environ["LDFLAGS"] = " -lmpiP -ldwarf -lelf"
+        targets.append('CXX = {0} {1}'.format(spack_cxx, ' -DUSE_MPI=0 '))
```

Comment:
This should probably only be added when `'~mpi' in self.spec`. Otherwise you'll be setting `CXX` twice.